main.js

```diff
@@ -1,67 +1,125 @@
+    });
+
+    document.getElementById('sortBtn').addEventListener('click', () => {
+        const sortedProducts = [...state.filteredProducts].sort((a, b) => a.price - b.price);
+        updateProductGrid(sortedProducts);
+    });
+
+    document.getElementById('resetBtn').addEventListener('click', resetFilters);
+
+    // Price range filter
+    elements.maxPrice.addEventListener('input', handlePriceFilter);
+}
+
+// Cart functionality
+function handleAddToCart() {
+    state.cartCount++;
+    localStorage.setItem(CONFIG.STORAGE_KEY, state.cartCount);
+    updateCart();
+    showNotification('Item added to cart!');
+}
+
 function updateCart() {
-    document.getElementById('count').textContent = cartCount;
+    elements.cartCount.textContent = state.cartCount;
 }
 
-// Show notification
-function showNotification() {
-    const notification = document.getElementById('notification');
-    notification.style.display = 'block';
+// Notification system
+function showNotification(message) {
+    elements.notification.textContent = message;
+    elements.notification.style.display = 'block';
+    
     setTimeout(() => {
-        notification.style.display = 'none';
-    }, 2000);
+        elements.notification.style.display = 'none';
+    }, CONFIG.NOTIFICATION_DURATION);
+}
+
+// Product filtering
+function filterProducts(filterFn) {
+    state.filteredProducts = state.products.filter(filterFn);
+    updateProductGrid(state.filteredProducts);
 }
 
-// Sort products
-document.getElementById('sortBtn').addEventListener('click', () => {
-    const sortedProducts = [...products].sort((a, b) => a.price - b.price);
-    updateProductGrid(sortedProducts);
-});
+function handlePriceFilter(e) {
+    const maxPrice = parseInt(e.target.value);
+    state.maxPrice = maxPrice;
+    elements.priceValue.textContent = `$${maxPrice}`;
+    filterProducts(product => product.price <= maxPrice);
+}
 
-// Filter products
-document.getElementById('filterBtn').addEventListener('click', () => {
-    const filteredProducts = products.filter(product => product.price < 100);
-    updateProductGrid(filteredProducts);
-});
+// Reset filters
+function resetFilters() {
+    state.maxPrice = CONFIG.DEFAULT_MAX_PRICE;
+    elements.maxPrice.value = CONFIG.DEFAULT_MAX_PRICE;
+    elements.priceValue.textContent = `$${CONFIG.DEFAULT_MAX_PRICE}`;
+    state.filteredProducts = [...state.products];
+    updateProductGrid(state.filteredProducts);
+}
 
-// Update product grid
+// Grid updates
 function updateProductGrid(products) {
-    const grid = document.getElementById('productGrid');
-    grid.innerHTML = products.map(product => `
-        <div class="card">
-            <h3>${product.name}</h3>
-            <p>Price: $${product.price}</p>
-            <button class="btn add-to-cart">Add to Cart</button>
-        </div>
-    `).join('');
+    const productHTML = products
+        .map(({ name, price, inStock }) => `
+            <div class="card">
```

Comment:
```suggestion
            <div class="card" data-product-id="${id}">
```

Add data attribute for product ID - this will make it easier to track which product was clicked without relying on DOM structure.